server/data/comments.js

```diff
@@ -1,9 +1,47 @@
 const { ObjectId } = require('mongodb');
 const { getProjectById } = require('./projects');
 const { projects } = require('../config/mongoCollections');
-const { isValidStr, internalServerErr, isValidObjectId } = require('../utils');
+const {
+	isValidStr,
+	internalServerErr,
+	isValidObjectId,
+	notFoundErr,
+	forbiddenErr,
+} = require('../utils');
 const { isValidUsername } = require('../utils/users');
 
+const getCommentById = async (commentId) => {
+	const id = isValidObjectId(commentId);
+	const projectsCollection = await projects();
+	const comment = await projectsCollection.findOne({
+		comments: { $elemMatch: { _id: ObjectId(id) } },
+	});
+	if (!comment) throw notFoundErr('No comment found for the provided id');
+	return comment;
+};
+
+const iscommentOwner = async (projectId, commentId, userId) => {
+	let isOwner = false;
+	projectId = isValidObjectId(projectId);
+	commentId = isValidObjectId(commentId);
+	userId = isValidObjectId(userId.toString());
+	const projectFind = await getProjectById(projectId);
+	if (projectFind.owner._id.toString() === userId) {
+		isOwner = true;
```

Comment:
i think you can return `true` from here, if the user is project owner then we don't need to go ahead and iterate over all the comments

Reply:
Done.